lib/engine/step/g_1849/token.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require_relative '../token'
+
+module Engine
+  module Step
+    module G1849
+      class Token < Token
+        def process_place_token(action)
+          entity = action.entity
+
+          place_token(entity, action.city, action.token)
+
+          index = @game.corporations.index { |c| c.name == 'AFG' }
+          afg = index ? @game.corporations[index] : nil
```

Comment:
why don't you just do @game.corporations.find ...

Reply:
I need the index on line 18.

Reply:
then afg = @game.corporations[index] is the simplification

Reply:
isn't `array[nil]` a `TypeError`?

Reply:
To clarify, AFG may not be in the game.

Reply:
correct, i'm mistaken